pkg/apis/datadoghq/v1alpha1/datadogagentdeployment_types.go

```diff
@@ -114,6 +114,14 @@ type DatadogAgentDeploymentSpecAgentSpec struct {
 	// SystemProbe configuration
 	// +optional
 	SystemProbe SystemProbeSpec `json:"systemProbe,omitempty"`
+
+	// Confd configuration
+	// +optional
+	Confd *DirConfig `json:"confd,omitempty"`
+
+	// Checksd configuration
```

Comment:
could you add more documentation here as well ? https://docs.datadoghq.com/developers/write_agent_check